_helper/post-upgrade-config.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+# fail on error, unassigned var, pipe error
+set -euo pipefail
+
+CONF_DIR="$1"
+# Next paths are configurable, getting them as params
+RESTIC_PASSWORD_PATH="$2"
+BACKUP_EXCLUSIONS_PATH="$3"
+
+declare -A config_values
+
+COMMENTED_LINE_PATTERN="^[[:blank:]]*#"
+
+# --vvv-- PARSE/STORE CONFIGS from backup files --vvv--
+
+# Skip comments and take assignments only (key/value pairs).
+filter_var_assignments() {
+	local input_file="$1"
+	echo "$(grep -v "$COMMENTED_LINE_PATTERN" "$input_file" | egrep '\w+\+?=')"
+}
+
+# Remove leading and trailing stuff from var assignment, like exports and trailing comments.
+trim_kv_pair() {
+	line="$1"
+	# considers patterns: key="VALUE" and key+="VALUE"
+	echo "$(sed -E 's|.*(\b\w+\b\+?=".*").*|\1|' <<< "$line")"
+}
+
+store_env_vars() {
+	local config_file_backup="$1"
+
+	local kv_lines="$(filter_var_assignments "$config_file_backup")"
```

Comment:
This could work for simple cases. We have to keep in mind that the env files are shell scripts and someone could do something like building up a value over a few statements and with different syntax like

```bash
RESTIC_BACKUP_PATHS="$HOME/dir"
if [...]; then
   RESTIC_BACKUP_PATHS+=" /something"
else
   RESTIC_BACKUP_PATHS="${RESTIC_BACKUP_PATHS} /other"
end
```

Correctly finding the values for all cases I think is quite hard

Maybe a safer alternative would be to 
1. source the `.env.sh` file
2. store the values now available in envvars like `$RESTIC_BACKUP_PATHS`
3. Start from a new clean config and insert those values

Hmm actually this is also not perfect, as a typical thing could be

```bash
if [ -d /mnt/some_disk ] ; then
   RESTIC_BACKUP_PATHS+=" /mnt/some_disk"
end
```
and the value depends on how the system looked at this time.